### ISSUE TITLE
Showing that Data.Nat.Show.showInBase is injective

### ISSUE THREAD

**Issue by umazalakain** [agda/agda-stdlib#1248]
Would it be interesting to add a proof that shows that Data.Nat.Show.showInBase is injective? I have such a proof for base 10, but I could generalize. This would require https://github.com/agda/agda/pull/4790 to be merged into Agda. Depends on #1256 #1255 and #1251.

**Comment by gallais** [comment]
That looks good. You need to run `make interaction` to fix the
test case using search-about to find all the definitions mentioning
`String` because the new primitive is mentioned in the output now!

**Comment by gallais** [comment]
Sounds good. Your PR should target the experimental branch if it
relies on an unreleased feature.

**Comment by umazalakain** [comment]
Closing this as `showInBase` is now split into `charsInBase` and `String.fromList`: the former has been shown to be injective and the injectivity the latter has made it into Agda's master branch.

https://github.com/agda/agda-stdlib/commit/b52e3e5d6185f119211cfdf8c63d1a0f9b2502a8
